src/runtime_src/core/common/info_aie.cpp

```diff
@@ -0,0 +1,701 @@
+    rtp.put("selector_column", rtp_node.second.get<uint16_t>("selector_column"));
+    rtp.put("selector_lock_id", rtp_node.second.get<uint16_t>("selector_lock_id"));
+    rtp.put("selector_address", rtp_node.second.get<uint64_t>("selector_address"));
+
+    rtp.put("ping_buffer_row", rtp_node.second.get<uint16_t>("ping_buffer_row"));
+    rtp.put("ping_buffer_column", rtp_node.second.get<uint16_t>("ping_buffer_column"));
+    rtp.put("ping_buffer_lock_id", rtp_node.second.get<uint16_t>("ping_buffer_lock_id"));
+    rtp.put("ping_buffer_address", rtp_node.second.get<uint64_t>("ping_buffer_address"));
+
+    rtp.put("pong_buffer_row", rtp_node.second.get<uint16_t>("pong_buffer_row"));
+    rtp.put("pong_buffer_column", rtp_node.second.get<uint16_t>("pong_buffer_column"));
+    rtp.put("pong_buffer_lock_id", rtp_node.second.get<uint16_t>("pong_buffer_lock_id"));
+    rtp.put("pong_buffer_address", rtp_node.second.get<uint64_t>("pong_buffer_address"));
+
+    rtp.put("is_pl_rtp", rtp_node.second.get<bool>("is_PL_RTP"));
+    rtp.put("is_input", rtp_node.second.get<bool>("is_input"));
+    rtp.put("is_asynchronous", rtp_node.second.get<bool>("is_asynchronous"));
+    rtp.put("is_connected", rtp_node.second.get<bool>("is_connected"));
+    rtp.put("requires_lock", rtp_node.second.get<bool>("requires_lock"));
+    rtp_array.push_back(std::make_pair(rtp_node.first, rtp));
+  }
+
+  pt_array.add_child("rtps", rtp_array);
+}
+
+// Populate GMIOs for AIE Cores
+void
+populate_aie_core_gmio(boost::property_tree::ptree pt, boost::property_tree::ptree& pt_array)
+{
+  boost::property_tree::ptree gmio_array;
+  for (auto& gmio_node : pt.get_child("aie_metadata.GMIOs")) {
+    boost::property_tree::ptree gmio;
+    gmio.put("id", gmio_node.second.get<std::string>("id"));
+    gmio.put("name", gmio_node.second.get<std::string>("name"));
+    gmio.put("logical_name", gmio_node.second.get<std::string>("logical_name"));
+    gmio.put("type", gmio_node.second.get<uint16_t>("type"));
+    gmio.put("shim_column", gmio_node.second.get<uint16_t>("shim_column"));
+    gmio.put("channel_number", gmio_node.second.get<uint16_t>("channel_number"));
+    gmio.put("stream_id", gmio_node.second.get<uint16_t>("stream_id"));
+    gmio.put("burst_length_in_16byte", gmio_node.second.get<uint16_t>("burst_length_in_16byte"));
+    gmio.put("pl_port_name", gmio_node.second.get<std::string>("PL_port_name","N/A"));
+    gmio.put("pl_parameter_name", gmio_node.second.get<std::string>("PL_parameter_name","N/A"));
+    gmio_array.push_back(std::make_pair(gmio_node.first, gmio));
+  }
+
+  pt_array.add_child("gmios",gmio_array);
+}
+
+boost::property_tree::ptree
+populate_aie(const xrt_core::device *device, const std::string& desc)
+{
+  boost::property_tree::ptree pt;
+  std::string aie_data;
+  std::string aie_core_data;
+  std::vector<std::string> graph_status;
+  pt.put("description", desc);
+  boost::property_tree::ptree graph_array;
+  boost::property_tree::ptree _pt;
```

Comment:
Could you rename "pt" and "_pt" to be a more obvious property tree name.

Reply:
Done. Thanks